journal/src/main/java/io/camunda/zeebe/journal/file/JournalMetrics.java

```diff
@@ -104,16 +104,20 @@ final class JournalMetrics {
   private final Counter.Child appendRate;
   private final Counter.Child appendDataRate;
 
-  JournalMetrics(final String logName) {
-    segmentCreationTime = SEGMENT_CREATION_TIME.labels(logName);
-    segmentTruncateTime = SEGMENT_TRUNCATE_TIME.labels(logName);
-    segmentFlushTime = SEGMENT_FLUSH_TIME.labels(logName);
-    segmentCount = SEGMENT_COUNT.labels(logName);
-    journalOpenTime = JOURNAL_OPEN_DURATION.labels(logName);
-    segmentAllocationTime = SEGMENT_ALLOCATION_TIME.labels(logName);
-    appendLatency = APPEND_LATENCY.labels(logName);
-    appendRate = APPEND_RATE.labels(logName);
-    appendDataRate = APPEND_DATA_RATE.labels(logName);
+  JournalMetrics(final String name) {
+    // hacky way of fetching the partition ID if the journal was created from a partition group; if
+    // not, the name is used as is
+    final var partition = name.substring(name.lastIndexOf('-') + 1);
```

Comment:
:wrench: Can't we "just" pass in the partition id? `RaftPartitionServer#createRaftStorage` knows the id, from there it's just a lot of passing it down...

Reply:
We can. Keep in mind the journal name is not the ID, but [`partitionGroupName`-partition-`partitionId`]

So passing the partition ID here is really just for the metrics and nothing else. It felt like more refactoring than I was comfortable doing for this, since the storage is built in a way that it has a prefix and is not partition aware really. I'm not against pushing down to the storage the building of the prefix/name thing, but I also didn't see a ton of value of doing it right now :shrug: 

Doing it, means we have to refactor the `RaftStorageBuilder` to take in the partition group and the ID, which then creates the prefix itself and owns it (since it needs it for the lock and other stuff). We then pass it down to the journal (for backwards compat) but also the ID just for the metrics. Seems like a lot of cruft for just that :smile: Though maybe that's saying more about the whole builder set up than anything else :smile: 

I'm happy to do that. There's some small value in being explicit, for sure, just seemed to me like very small :sweat_smile: 

I added a last commit which does exactly this. If you prefer that, then let's go with it.

Reply:
Thanks for doing this! I do prefer it this way but feel free to just reject a :wrench: next time :wink:

Reply:
Ofc, but in this case it was a pretty quick/small change anyway.